### ISSUE TITLE
False positive no-member when concatenating tuples

### ISSUE THREAD

**Issue by mariojonke** [PyCQA/pylint#4098]
### Steps to reproduce
1. Run pylint on the following sample code:
```python
# pylint:disable=missing-module-docstring
# pylint:disable=missing-class-docstring
# pylint:disable=missing-function-docstring
# pylint:disable=too-few-public-methods

import typing


class Value:
    def print_value(self):
        pass


class MultiValue(Value):
    def __init__(self):
        self._values = ()  # type: typing.Tuple[Value, ...]

    def add_value(self, value: Value):
        self._values = self._values + (value, )

    def print_value(self):
        for value in self._values:
            value.print_value()
```

### Current behavior
```
************* Module my_file
my_file.py:23:12: E1101: Instance of 'tuple' has no 'print_value' member (no-member)
```

### Expected behavior
No `no-member` linting erros.

### pylint --version output
```
pylint 2.6.0
astroid 2.5
Python 3.8.0 (default, Oct 28 2019, 16:14:01) 
[GCC 9.2.1 20191008]
```
The problem seems to be related to `astroid` in version `2.5`. Downgrading to version `2.4.2` does not produce such `no-member` errors.

**Comment by mbyrnepr2** [comment]
This issue appears to be fixed. Tested on:
```
pylint 2.14.0-b1
astroid 2.11.5
Python 3.10.0b2
```
